### PR TITLE
iso8859-1: initial implementation to the `is_numeric` method and a minor improvement to the test

### PR DESCRIPTION
**Pull request by danielgoncalvesti**

### CODE REVIEW

File: iso8859-1/src/lib.rs [hunk 1]
```diff
@@ -139,7 +139,7 @@ impl IsoLatin1Char {
     /// TODO
     /// ```
     pub fn is_numeric(&self) -> bool {
-        todo!()
+        self >= &IsoLatin1Char(0x30) && self <= &IsoLatin1Char(0x39)
```
Comment (GrayJack) [inline iso8859-1/src/lib.rs]: There is a better syntax for this. You can get the anonymous field using `self.0`, e.g. `self.0 >= 0x30`

Comment (GrayJack) [inline iso8859-1/src/lib.rs]: This implementation is incomplete. These standards don't specify the classification, but the Unicode standard specifies such classification for the value. Since those classifications and methods are very convenient, we can use Unicode classification for numbers to classify the bytes representing these Unicode values. That means that superscripts like "²" and fractions like "½" are also numeric digits.

File: iso8859-1/src/lib.rs [hunk 2]
```diff
@@ -344,7 +344,9 @@ mod api_tests {
         for byte in b'0'..=b'9' {
             assert!(IsoLatin1Char(byte).is_numeric());
         }
-        todo!()
+        for byte in b'a'..=b'z' {
+            assert!(!IsoLatin1Char(byte).is_numeric());
+        }
```
Comment (GrayJack) [inline iso8859-1/src/lib.rs]: That is very good, but also don't forget to add tests for the second part of the character table as well

**Review by GrayJack — CHANGES_REQUESTED** [review]

**Comment by silasdb** [comment]
On Tue, Jul 26, 2022 at 10:52:14AM -0700, Eric Shimizu Karbstein wrote:
> > @@ -139,7 +139,7 @@ impl IsoLatin1Char {
>      /// TODO
>      /// ```
>      pub fn is_numeric(&self) -> bool {
> -        todo!()
> +        self >= &IsoLatin1Char(0x30) && self <= &IsoLatin1Char(0x39)
> 
> This implementation is incomplete. These standards don't specify the classification, but the Unicode standard specifies such classification for the value. Since those classifications and methods are very convenient, we can use Unicode classification for numbers to classify the bytes representing these Unicode values. That means that superscripts like "²" and fractions like "½" are also numeric digits.

Hi. Since we are dealing with ISO-8859-1 (Latin-1). Does it make sense to talk about Unicode numeric digits like fractions?

Also, instead of using 0x30 .. 0x39, is it a better approach to use rust syntax b'0' ... b'9'?

**Comment by GrayJack** [comment]
> Hi. Since we are dealing with ISO-8859-1 (Latin-1). Does it make sense to talk about Unicode numeric digits like fractions?

It does, for some reason, there is no other big openly known classification of character symbols, and most if not all of those old standards do not specify that, if we let that implementation-defined we will see each sub crate using one definition and the last but not least, those symbols do represent something related to numerics so it makes sense those being included.

In my personal opinion, we don't need to follow that definition, as long as we get to a well-defined what means to be a numeric symbol and it be used as a definition on all encodes that we do, doesn't matter what symbol is considered or not.

> Also, instead of using 0x30 .. 0x39, is it a better approach to use rust syntax b'0' ... b'9'?

It's ok to use either way since those are still on the part where this particular standard is the same as the ASCII code points. In the encoding I got, I'm using on the implementation site only the `0x` syntax for consistency with the code dealing with codepoints after ASCII

**Comment by silasdb** [comment]
On Tue, Jul 26, 2022 at 07:12:19PM -0700, Eric Shimizu Karbstein wrote:
> > Hi. Since we are dealing with ISO-8859-1 (Latin-1). Does it make sense to talk about Unicode numeric digits like fractions?
> 
> It does, for some reason, there is no other big openly known classification of character symbols, and most if not all of those old standards do not specify that, if we let that implementation-defined we will see each sub crate using one definition and the last but not least, those symbols do represent something related to numerics so it makes sense those being included.
> 
> In my personal opinion, we don't need to follow that definition, as long as we get to a well-defined what means to be a numeric symbol and it be used as a definition on all encodes that we do, doesn't matter what symbol is considered or not.

Sorry, I think I didn't get it yet. ISO-8859-1 (Latin-1) is a one-byte character
set. "½" only makes sense in a multibyte character set. In UTF-8 it is encoded
as C2 BD. How one would compare a numeric digit in Latin-1 (one byte: '0': 0x30,
'1': 0x31, etc.) with other numbers defined in UTF-8 which encodes "½" as 0xC2
0xBD?

In other words, it is impossible to represent "½" in Latin-1, so for me it is
not clear how we should perform such tests.

**Comment by silasdb** [comment]
On Wed, Jul 27, 2022 at 10:40:21AM -0300, Silas wrote:
> Sorry, I think I didn't get it yet. ISO-8859-1 (Latin-1) is a one-byte character
> set. "½" only makes sense in a multibyte character set. In UTF-8 it is encoded
> as C2 BD. How one would compare a numeric digit in Latin-1 (one byte: '0': 0x30,
> '1': 0x31, etc.) with other numbers defined in UTF-8 which encodes "½" as 0xC2
> 0xBD?
> 
> In other words, it is impossible to represent "½" in Latin-1, so for me it is
> not clear how we should perform such tests.

Nevermind, "¼", "½", "¾" and others are defined in ISO-8859-1 (Latin-1).

https://en.wikipedia.org/wiki/ISO/IEC_8859-1

Thanks!

File: iso8859-1/src/lib.rs (hunk 2 above)
Comment (danielgoncalvesti) [reply iso8859-1/src/lib.rs]: @GrayJack good point! I will create an array with all numeric values, and then I will test them inside the for-loop.

**Review by GrayJack — APPROVED** [review]
Looks good to me
